flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -887,20 +887,28 @@ private ApplicationReport startAppMaster(
 			}
 		}
 
-		// setup security tokens
 		Path remotePathKeytab = null;
+		String localizedKeytabPath = null;
 		String keytab = configuration.getString(SecurityOptions.KERBEROS_LOGIN_KEYTAB);
 		if (keytab != null) {
-			LOG.info("Adding keytab {} to the AM container local resource bucket", keytab);
-			remotePathKeytab = setupSingleLocalResource(
-					Utils.KEYTAB_FILE_NAME,
+			boolean	requireLocalizedKeytab = flinkConfiguration.getBoolean(YarnConfigOptions.SHIP_LOCAL_KEYTAB);
+			localizedKeytabPath = flinkConfiguration.getString(YarnConfigOptions.LOCALIZED_KEYTAB_PATH);
+			if (requireLocalizedKeytab) {
```

Comment:
I think this can be moved to the outer if, i.e. `keytab != null && requireLocalizedKeytab`

Reply:
+1